tools/runner.py

```diff
@@ -0,0 +1,181 @@
+        leaderboard[item][4] = tournament_score(leaderboard[item])
+
+    leaderboard = sorted(leaderboard.items(), reverse=True, key=lambda itm: itm[1][4])
+
+    print("{:6s} {:25s} {:7s} | {:23s} | {:23s} | {:23s} | {:23s}".format("", "", "", "Starter Rounds", "Second Rounds", "Starter matches", "Second matches"))
+
+    print("{:6s} {:25s} {:7s} | {:7s} {:7s} {:7s} | {:7s} {:7s} {:7s} | {:7s} {:7s} {:7s} | {:7s} {:7s} {:7s}".format("Place", "Name", "Score", "Wins", "Draws", "Losses", "Wins", "Draws", "Losses", "Wins", "Draws", "Losses", "Wins", "Draws", "Losses"))
+    idx = 0
+    for item in leaderboard:
+        idx += 1
+        print(
+            "{:6d} {:25s} {:7d} | {:7d} {:7d} {:7d} | {:7d} {:7d} {:7d} | {:7d} {:7d} {:7d} | {:7d} {:7d} {:7d}".
+                format(idx, item[0], item[1][4],
+                       item[1][0][0], item[1][0][1], item[1][0][2],
+                       item[1][1][0], item[1][1][1], item[1][1][2],
+                       item[1][2][0], item[1][2][1], item[1][2][2],
+                       item[1][3][0], item[1][3][1], item[1][3][2]
+                       ))
+    pass
+
+
+def tournament_score(score):
+    # Score = Number of wins -  Number of losses
+    return (score[0][0] - score[0][2]) + (score[1][0] - score[1][2])
+
+
+def tournament(models):
+    players = []
+    for model in models:
+        players.append(tournament_player_loader(model))
+
+    print("Loaded", len(players), "players :", ", ".join([player.name for player in players]))
+
+    game_list = []
+    leaderboard = {}
+
+    for player1 in players:
+        leaderboard[player1.name] = [[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], 0]
+        for player2 in players:
+            if player1 == player2:
+                continue
+            game_list.append((player1, player2))
+
+    random.shuffle(game_list)
+
+    for game in game_list:
+        score = play_game(game[0], game[1], ROUNDS)
+        print(score)
+
+        p1 = game[0].name
+        p2 = game[1].name
+        leaderboard[p1][0] = [x + y for x, y in zip(leaderboard[p1][0], score)]
+        leaderboard[p2][1] = [x + y for x, y in zip(leaderboard[p2][1], score[::-1])]
+        if score[0] > score[2]:
+            leaderboard[p1][2][0] += 1
+            leaderboard[p2][3][2] += 1
+        elif score[0] == score[2]:
+            leaderboard[p1][2][1] += 1
+            leaderboard[p2][3][1] += 1
+        else:
+            leaderboard[p1][2][2] += 1
+            leaderboard[p2][3][0] += 1
+
+        if DISPLAY_LEADERBOARD_EACH_ROUND:
+            tournament_print(leaderboard)
+
+    if not DISPLAY_LEADERBOARD_EACH_ROUND:
+        tournament_print(leaderboard)
+
+    for player in players:
+        player.save_model()
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Tournament runner for connect four")
+    parser.add_argument("models", metavar="model", type=str, nargs="+", help="List of models")
+    args = parser.parse_args()
+    if len(args.models) < 2:
```

Comment:
This would exit even if I would pass it the model as argument. Probably the description/help should be changed accordingly to the validation conditions.

Reply:
> This would exit even if I would pass it the model as argument. Probably the description/help should be changed accordingly to the validation conditions.

The code is right, the description should be fixed as it expects at leas two models.